src/osd/PG.cc

```diff
@@ -3914,6 +3921,22 @@ void PG::do_delete_work(ObjectStore::Transaction &t)
     &next);
   dout(20) << __func__ << " " << olist << dendl;
 
+  if (_next) {
```

Comment:
i don't follow on this part. if `next` == `ghobject_t::get_max()`, then we hit the last object in the collection and there is no need to list the collection again, right?

Reply:
@tchaikov - this is a bit paranoic check intended to make sure no new entries has appeared in the PG since it was scheduled for removal. 
Original removal procedure was resistant to such issue since it started enumeration from the beginning on each step. Hence even if new entries appeared they are to be removed properly.
New procedure isn't that reliable due to next position usage. Hence decided to introduce this additional checking.

Reply:
@ifed01 I think this paranoid check is very nice addition to deletion procedure.
How about printing some derr in this case?